src/navigations/HomeBottomTabs.js

```diff
@@ -207,135 +211,116 @@ function HomeBottomTabs({navigation}) {
     }
   }, [initialStartup, otherProfileData]);
 
-  const renderTabLabelIcon =
-    (componentType) =>
-    // eslint-disable-next-line react/display-name
-    ({color}) => {
-      if (componentType === 'Feed') {
-        return (
-          <View style={styles.center}>
-            <MemoFeed fill={color} />
-          </View>
-        );
-      }
-      if (componentType === 'ChannelList') {
-        return (
-          <View style={styles.center}>
-            <MemoHome fill={color} />
-          </View>
-        );
-      }
-      if (componentType === 'News') {
-        return (
-          <View>
-            <MemoNews fill={color} />
-          </View>
-        );
-      }
-
+  const renderTabBarIcon = (route, focused, color) => {
+    if (route.name === 'SignedChannelList') {
       return (
         <View style={styles.center}>
-          <MemoProfileIcon />
+          <SignedChat fill={color} stroke={color} />
         </View>
       );
-    };
-  // eslint-disable-next-line react/display-name
+    }
+    if (route.name === 'AnonymousChannelList') {
+      return (
+        <View style={styles.center}>
+          {focused ? (
+            <AnonymousChatFill fill={colors.anon_primary} stroke={colors.anon_primary} />
+          ) : (
+            <AnonymousChatOutline fill={color} />
+          )}
+        </View>
+      );
+    }
+    if (route.name === 'Feed') {
+      return (
+        <View style={styles.center}>
+          <MemoFeed fill={color} />
+        </View>
+      );
+    }
+    if (route.name === 'News') {
+      return (
+        <View>
+          <MemoNews fill={color} />
+        </View>
+      );
+    }
+
+    return (
+      <View style={styles.center}>
+        <MemoProfileIcon />
+      </View>
+    );
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `renderTabBarIcon` function, you're checking the route name multiple times with if statements. This could be simplified and made more maintainable by using a switch statement or an object mapping instead. Here's how you could do it with an object mapping:

```javascript
const iconComponents = {
  'SignedChannelList': <SignedChat fill={color} stroke={color} />,
  'AnonymousChannelList': focused ? 
    <AnonymousChatFill fill={colors.anon_primary} stroke={colors.anon_primary} /> : 
    <AnonymousChatOutline fill={color} />,
  'Feed': <MemoFeed fill={color} />,
  'News': <MemoNews fill={color} />,
  'default': <MemoProfileIcon />
};

const renderTabBarIcon = (route, focused, color) => (
  <View style={styles.center}>
    {iconComponents[route.name] || iconComponents['default']}
  </View>
);
```

This way, if you need to add more icons in the future, you just need to add them to the `iconComponents` object.



<!-- This is an auto-generated comment by OSS CodeRabbit -->